packages/astro/src/core/render/params-and-props.ts

```diff
@@ -47,10 +47,11 @@ export async function getProps(opts: GetParamsAndPropsOptions): Promise<Props> {
 		base,
 	});
 
+	if (!staticPaths.length) return {};
+	const params = getParams(route, decodeURI(pathname));
 	// The pathname used here comes from the server, which already encored.
 	// Since we decided to not mess up with encoding anymore, we need to decode them back so the parameters can match
 	// the ones expected from the users
-	const params = getParams(route, decodeURI(pathname));
```

Comment:
There's no need to move this line, especially because the comments belongs exactly this line you just moved. This will make the comments confusing. Please revert this tiny change.